### PR TITLE
feat(npm): add a postinstall script to the cli package

### PR DESCRIPTION
**Pull request by leops**
## Summary

Add a `postinstall` script to the `rome` npm package to verify the installation of the native binary, and print a warning to the user immediately if something went wrong.

I decided to only print a warning when something is wrong instead of exiting with an error so that it won't cause an `npm install` on a whole workspace where `rome` is a dependency to fail, in case the user didn't intend to use the Rome CLI anyway.

### CODE REVIEW

File: npm/rome/scripts/postinstall.js [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+};
+
+const binName = PLATFORMS?.[platform]?.[arch];
+if (binName) {
+	let binPath;
+	try {
+		binPath = require.resolve(binName);
+	} catch (err) {
+		console.warn(
+			`The Rome CLI postinstall script failed to resolve the binary file "${binName}". ` + "Running Rome from the npm package will probably not work correctly.",
```
Comment (MichaReiser) [inline npm/rome/scripts/postinstall.js:25]: Interesting formatting...

File: npm/rome/scripts/postinstall.js [hunk 2]
```diff
@@ -0,0 +1,44 @@
+const { platform, arch } = process;
+
+const PLATFORMS = {
```
Comment (MichaReiser) [inline npm/rome/scripts/postinstall.js:3]: Do we still need the `bin/rome` file. This file looks very much the same?

Comment (leops) [reply npm/rome/scripts/postinstall.js:3]: The logic in both files is mostly the same, the difference being that the postinstall script only prints a warning while the wrapper script in `bin/rome` prints an error and exit with a non-zero status code if an error happens

File: npm/rome/scripts/postinstall.js (hunk 1 above)
Comment (leops) [reply npm/rome/scripts/postinstall.js:25]: That's a bug in how binary expressions are formatted, they don't create a new group if there is less than three items but that doesn't account for the size of these sub-expressions

File: npm/rome/scripts/postinstall.js [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+		binPath = require.resolve(binName);
+	} catch (err) {
+		console.warn(
+			`The Rome CLI postinstall script failed to resolve the binary file "${binName}". ` + "Running Rome from the npm package will probably not work correctly.",
+		);
+	}
+
+	if (binPath) {
+		try {
+			require("fs").chmodSync(binPath, 0o755);
```
Comment (ematipico) [inline npm/rome/scripts/postinstall.js:31]: Let's use `node:fs`

**Review by ematipico — APPROVED** [review]
Looks good to me, just a minor comment.

**Pull request merged**